[2.4.0] feat: use .ls-lint.yaml file as default configuration file

ccoVeille opened the pull request:
The .ls-lint.yaml file is now used as the default configuration file
The .ls-lint.yml file is still supported for backward compatibility.

The choice was made to prioritize the .ls-lint.yaml file for its clarity
and alignment with YAML file naming conventions.

Fixes #328

loeffel-io reviewed (changes requested):
you can easily remove the comments and consts please and then its good to go

background: the legacy config file will be deprecated and removed anyways in a period of time

ccoVeille commented:
@loeffel-io I simplified things

loeffel-io commented on cmd/ls_lint/main.go:
```diff
@@ -57,7 +57,19 @@ func main() {
 
 	if len(flagConfig) == 0 {
-		flagConfig = _flag.Config{".ls-lint.yml"}
+		// no config files were provided by the --config flag
```
```suggestion
```

loeffel-io commented on README.md:
```diff
@@ -11,8 +11,8 @@ An extremely fast directory and filename linter - Bring some structure to your p
 
-- Minimal setup with simple rules managed in one single or multiple `.ls-lint.yml` files
-- Works for directory and file names - all extensions supported - full unicode support
+- Minimal setup with simple rules managed in one single or multiple `.ls-lint.yaml` (or `.ls-lint.yml`) files
```
```suggestion
- Minimal setup with simple rules managed in one single or multiple `.ls-lint.yaml` files
```

loeffel-io reviewed (changes requested):
LGTM, i think we can just rm this

ccoVeille committed:
feat: use .ls-lint.yaml file as default configuration file

The .ls-lint.yaml file is now used as the default configuration file
The .ls-lint.yml file is still supported for backward compatibility.

The choice was made to prioritize the .ls-lint.yaml file for its clarity
and alignment with YAML file naming conventions.

loeffel-io commented:
also https://ls-lint.org/2.3/getting-started/installation.html#run-from-source could need a update for 2.4
but i can do this myself, needs a bit more work
